magicctapipe/scripts/lst1_magic/setting_up_config_and_dir.py

```diff
@@ -0,0 +1,466 @@
+    "\n        picture_thresh: 8",
+    "\n        boundary_thresh: 4",
+    "\n        keep_isolated_pixels: false",
+    "\n        min_number_picture_neighbors: 2",
+    "\n",
+    "\n    time_delta_cleaning:",
+    "\n        use: true",
+    "\n        min_number_neighbors: 1",
+    "\n        time_limit: 2",
+    "\n",
+    "\n    dynamic_cleaning:",
+    "\n        use: true",
+    "\n        threshold: 267",
+    "\n        fraction: 0.03",
+    "\n",
+    "\n    use_only_main_island: false",
+    "\n",
+    "\nMAGIC:",
+    "\n    image_extractor:",
+    '\n        type: "SlidingWindowMaxSum"',
+    "\n        window_width: 5",
+    "\n        apply_integration_correction: false",
+    "\n",
+    "\n    charge_correction:",
+    "\n        use: true",
+    "\n        factor: 1.143",
+    "\n",
+    "\n    magic_clean:",
+    "\n        use_time: true",
+    "\n        use_sum: true",
+    "\n        picture_thresh: 6",
+    "\n        boundary_thresh: 3.5",
+    "\n        max_time_off: 4.5",
+    "\n        max_time_diff: 1.5",
+    "\n        find_hotpixels: true",
+    '\n        pedestal_type: "from_extractor_rndm"',
+    "\n",
+    "\n    muon_ring:",
+    "\n        thr_low: 25",
+    "\n        tailcut: [12, 8]",
+    "\n        ring_completeness_threshold: 25",
+    "\n"]
+    
+    f.writelines(lines_of_config_file)
+    f.close()
+
+
+def lists_and_bash_generator(particle_type, target_dir, MC_path, SimTel_version, focal_length, scripts_dir):
+
+    """
+    This function creates the lists list_nodes_gamma_complete.txt and list_folder_gamma.txt with the MC file paths.
+    After that, it generates a few bash scripts to link the MC paths to each subdirectory. 
+    These bash scripts will be called later in the main() function below. 
+    """
+    
+    process_name = target_dir.split("/")[-2:][1]
+    
+    list_of_nodes = glob.glob(MC_path+"/node*")
+    f = open(target_dir+f"/list_nodes_{particle_type}_complete.txt","w") # creating list_nodes_gammas_complete.txt
+    for i in list_of_nodes:
+        f.write(i+"/output_"+SimTel_version+"\n")   
+    
+    f.close()
+    
+    f = open(target_dir+f"/list_folder_{particle_type}.txt","w") # creating list_folder_gammas.txt
+    for i in list_of_nodes:
+        f.write(i.split("/")[-1]+"\n")   
+    
+    f.close()
+    
+    ####################################################################################
+    ############ bash scripts that link the MC paths to each subdirectory. 
+    ####################################################################################
+    
+    f = open(f"linking_MC_{particle_type}_paths.sh","w")
+    lines_of_config_file = [
+    "#!/bin/sh\n\n",
+    "#SBATCH -p short\n",
+    "#SBATCH -J "+process_name+"\n\n",
+    "#SBATCH -N 1\n\n",
+    "ulimit -l unlimited\n",
+    "ulimit -s unlimited\n",
+    "ulimit -a\n\n",
+    "while read -r -u 3 lineA && read -r -u 4 lineB\n",
+    "do\n",
+    "    cd "+target_dir+f"/DL1/MC/{particle_type}\n",
```

Comment:
here you are using string addition and f-strings. can be just and f-string, which would make the code also more readable (this applies to all places where you use string addition, f-string does the same but better!)

Reply:
Done